### PR TITLE
Defer constant loading of `ActionMailer::MailDeliveryJob` until ActiveJob and ActionMailer is loaded

### PR DESCRIPTION
**Pull request by bensheldon**
Sibling PR to #45476, where this was initially split off from.

Connects to #45434.

### CODE REVIEW

File: actionmailer/lib/action_mailer/base.rb [hunk 1]
```diff
@@ -486,7 +486,7 @@ class Base < AbstractController::Base
 
     helper ActionMailer::MailHelper
 
-    class_attribute :delivery_job, default: ::ActionMailer::MailDeliveryJob
+    class_attribute :delivery_job, default: nil
```
Comment (rafaelfranca) [inline actionmailer/lib/action_mailer/base.rb:489]: This should not be `nil` by default.

**Review by rafaelfranca — CHANGES_REQUESTED** [review]
A better fix would always accept strings in the configuration and only convert to a constant in the first time the value is used, and cache it.

**Comment by bensheldon** [comment]
Just wanted to note that I discovered there is a hard dependency between ActionMailer and ActiveJob, introduced in #16598, and necessary given the current implementation:

https://github.com/rails/rails/blob/c3c747547a85937e67bc62cfb9f466948fb631ad/actionmailer/lib/action_mailer/railtie.rb#L3

By using a String-deferral strategy, I think it's possible to drop the `require`.

**Comment by bensheldon** [comment]
Closing this for now.

**Pull request closed without merging**